sympy/functions/special/gamma_functions.py

```diff
@@ -1057,6 +1057,19 @@ def fdiff(self, argindex=1):
         else:
             raise ArgumentIndexError(self, argindex)
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy import symbols, Wild
+        z = self.args[0].as_leading_term(x, logx=logx, cdir=cdir)
+        z0 = z.subs(x, 0)
+        if z0.is_infinite:
+            r = -z*(-log(z) + 1)
+            if logx is not None:
+                c, ex = symbols('c, ex', cls=Wild, exclude=[x])
+                r = r.replace(log(c*x**ex), log(c) + ex*logx)
+                return r
+            return z*(-log(z))
+        return log(gamma(z))._eval_as_leading_term(x, logx=logx, cdir=cdir)
```

Comment:
This could probably be `return loggamma(z0)` as `gamma` is nonzero everywhere.

Reply:
Strictly speaking, we should also handle the poles of `gamma(z)` at `z = 0, -1, -2, ...` but that does not seem necessary for now.

Reply:
It can be changed to that when `logx` is None, otherwise it may give unexpected results